### DISCUSSION TITLE
How to manage an exception to `urgency.inherit=1`?

### DISCUSSION THREAD

**Discussion by allgreed** [GothenburgBitFactory/taskwarrior#3373] [Q&A]
Hi, I have this bit set up:
```
urgency.inherit=1
# this prevents funny business with dependency chains
urgency.blocking.coefficient=0
urgency.blocked.coefficient=0
```

and it works splendidly 99.99% of the time, because if something is blocking something with high urgency then it should have at least that urgency!
And then laundry happened. Basically laundry has some calculated urgency which I'm fine with. However there's a dependent task "hang laundry to dry" which needs to be done after laundry and for domain reasons (if I don't take it out of the washing machine it gets ugly) it warrants an urgency boost. However it should only be booster *after* the laundry is done. I've initially achieved this with just slapping `+next` on the drying task, however with the urgency inheritance it bumps the whole process to the top and that's not exactly what I want.

While writing this I've came to the realization that hooks might be the perfect use case for this -> namely add `+next` to the drying task once the laundry is completed, but no sooner. Is there a better way of approaching this issue?

**Comment by wrashi** [comment]
Hey @allgreed, I haven't tried crontrolling urgency like this before. Where did you get to? Did you write a hook?

**Reply by allgreed** [reply]
Yeah, I did, I'll put it under git soon and post a link here. It's not quite yet working well (I have an issue with running `tasklib` from within a hook) and it adds a noticeable (although not annoying) latency to each modification (since it has to be `on-modify-` hook).

The last issue could be probably solved by moving from Python to something compiled, dunno, will see ;) I'd greatly appreciate an `on-done-` hook though

**Reply by wrashi** [reply]
Sounds like good progress! Seems to be a general drift from Python to Rust with the project, so compilation may happen.

I don't think that anyone on the current dev team is looking at hooks. Perhaps you would consider joining the next bi-weekly meeting and making a contribution?
